src/interpreter/interpreter.cpp

```diff
@@ -272,14 +281,34 @@ struct ExpressionInterpreter : OverriddenVisitor<ExpressionInterpreter, Flow> {
 
 } // anonymous namespace
 
-std::vector<Literal> Interpreter::run(Expression* root) {
-  // Create a fresh store and execution frame, then run the expression to
-  // completion.
-  store = WasmStore();
-  store.callStack.emplace_back();
-  store.callStack.back().exprs = ExpressionIterator(root);
+Result<> Interpreter::addInstance(std::shared_ptr<Module> wasm) {
+  return instantiate(store.instances.emplace_back(wasm));
+}
 
+Result<> Interpreter::instantiate(Instance& instance) {
+  for (auto& global : instance.wasm->globals) {
+    store.callStack.emplace_back(instance, ExpressionIterator(global->init));
+    auto results = run();
+    instance.globalValues[global->name] = results[0];
```

Comment:
Is there something better I can do than assuming the first idx has the value of the global?

Reply:
Nope, since most globals will only have a single value. ("most" because we also support non-standard tuple locals, but we can cross that bridge later.) It would be good to add an assertion that `results` has size 1, but other than that, this look as good as it can get.